packages/runtime/src/observation/watcher-switcher.ts

```diff
@@ -0,0 +1,66 @@
+import { Collection } from '../observation';
+
+// todo:
+// merge collection subscription to property subscription
+// and make IWatcher simpler, so observers in static observation won't have to implement many methods
+// An alternative way is to make collection observation manual & user controllable
+// so it works even without proxy
+// todo: maybe enhance @connectable so that it provides these interfaces
+export interface IWatcher {
+  id: number;
+  observe(obj: object, property: PropertyKey): void;
+  observeCollection(collection: Collection): void;
+  observeLength(collection: Collection): void;
+}
+
+/**
+ * Current subscription collector
+ */
+let $watcher: IWatcher | null = null;
+const watchers: IWatcher[] = [];
+// eslint-disable-next-line
+export let watching = false;
+
+// todo: layer based collection pause/resume?
+export function pauseSubscription() {
+  watching = false;
+}
+
+export function resumeSubscription() {
+  watching = true;
+}
+
+export function currentWatcher(): IWatcher | null {
+  return $watcher;
+}
+
+export function enterWatcher(watcher: IWatcher): void {
+  if (watcher == null) {
+    throw new Error('watcher cannot be null/undefined');
+  }
+  if ($watcher == null) {
+    $watcher = watcher;
+    watchers[0] = $watcher;
+    watching = true;
```

Comment:
With `watching` being a number:
```suggestion
    ++watching;
```